tests/app/dao/test_services_dao.py

```diff
@@ -655,3 +658,90 @@ def test_dao_resume_service_marks_service_as_active_and_api_keys_are_still_revok
+        sample_template
+    )
+
+    notification_history(created_at=datetime(2016, 4, 1), status='sending')  # Start of financial year
+    notification_history(created_at=datetime(2016, 5, 30), status='created')
+    notification_history(created_at=datetime(2016, 6, 1), status='created')
+    notification_history(created_at=datetime(2017, 3, 31), status='created')  # End of financial year
+    notification_history(created_at=datetime(2017, 4, 1), status='created')
+    notification_history(created_at=datetime(2017, 5, 1), status='created')
+
+    result = dao_fetch_monthly_historical_stats_by_template_for_service(sample_template.service_id, 2016)
+
+    notifications_count = 0
+    for dict in result.values():
+        notifications_count += sum(dict.get(str(sample_template.id), {}).get("counts", {}).values())
+
+    assert '2017-04' not in result
+    assert '2017-05' not in result
+    assert notifications_count == 4
+
+
+def test_fetch_monthly_historical_template_stats_for_service_separates_months(
+    notify_db,
+    notify_db_session,
+    sample_template
+):
+    notification_history = functools.partial(
+        create_notification_history,
+        notify_db,
+        notify_db_session,
+        sample_template
+    )
+
+    notification_history(created_at=datetime(2016, 4, 1), status='sending')  # Start of financial year
+    notification_history(created_at=datetime(2016, 5, 30), status='created')
+    notification_history(created_at=datetime(2016, 6, 1), status='delivered')
+    notification_history(created_at=datetime(2016, 6, 1), status='created')
+    notification_history(created_at=datetime(2016, 12, 1), status='created')
+    notification_history(created_at=datetime(2017, 3, 30), status='sending')
+    notification_history(created_at=datetime(2017, 3, 31), status='sending')
+
+    result = dao_fetch_monthly_historical_stats_by_template_for_service(sample_template.service_id, 2016)
+
+    financial_year_month_keys = \
+        ['2016-{:02}'.format(month) for month in range(4, 13)] + ['2017-{:02}'.format(month) for month in range(1, 4)]
+
+    assert set(financial_year_month_keys) == set(result.keys())
+    assert sum(result["2016-04"][str(sample_template.id)]["counts"].values()) == 1
+    assert sum(result["2016-05"][str(sample_template.id)]["counts"].values()) == 1
+    assert sum(result["2016-06"][str(sample_template.id)]["counts"].values()) == 2
+    assert sum(result["2016-12"][str(sample_template.id)]["counts"].values()) == 1
+    assert sum(result["2017-03"][str(sample_template.id)]["counts"].values()) == 2
+
+
+def test_fetch_monthly_historical_template_stats_for_service_separates_templates(
+    notify_db,
+    notify_db_session
+):
+    notification_history = functools.partial(
+        create_notification_history,
+        notify_db,
+        notify_db_session,
+        status='delivered'
+    )
+
+    template_one = create_sample_template(notify_db, notify_db_session, template_id=uuid.uuid4())
```

Comment:
This appears to be the only place you are using the new template_id parameter of dao_create_template, but for what purpose? The change does not seem necessary